### PR TITLE
Chore: uFuzzy 1.0.2

### PR DESCRIPTION
**Pull request by leeoniya**
v1.0.1 pulls the `outOfOrder` boilerplate into a hugely optimized, new `uf.search()` API.
v1.0.2 adds multi term exclusions (e.g. `-blue -pink`): https://github.com/leeoniya/uFuzzy/issues/21

hopefully i didn't screw up the related refactors.

### CODE REVIEW

File: public/app/features/commandPalette/useMatches.ts [hunk 1]
```diff
@@ -201,35 +201,27 @@ function useInternalMatches(filtered: ActionImpl[], search: string): Match[] {
         const haystackItem = haystack[haystackIndex];
 
         // Use the position of the match as a stand-in for score
-        const substringPosition = haystackItem.toLowerCase().indexOf(query);
+        const substringPosition = haystackItem.indexOf(query);
 
         if (substringPosition > -1) {
-          const score = haystack.length - substringPosition;
```
Comment (leeoniya) [inline public/app/features/commandPalette/useMatches.ts:207]: @joshhunt  this looked like a mistake?

File: public/app/plugins/panel/flamegraph/components/FlameGraph/rendering.ts [hunk 2]
```diff
@@ -101,19 +101,17 @@ export function renderRect(
   const l = 65 + 7 * intensity;
 
   const name = rect.label;
-  const idxs = ufuzzy.filter([name], query);
```
Comment (leeoniya) [inline public/app/plugins/panel/flamegraph/components/FlameGraph/rendering.ts:104]: @joey-grafana it's much cheaper to do this filtering once on big haystack than 100k times on a haystacks of 1 item.

File: public/app/plugins/panel/flamegraph/components/FlameGraph/FlameGraph.tsx [hunk 3]
```diff
@@ -90,6 +90,27 @@ const FlameGraph = ({
     [levels, totalTicks, rangeMin]
   );
 
+  const [ufuzzy] = useState(() => {
+    return new uFuzzy();
+  });
+
+  // TODO: only run on first search of new data
```
Comment (joey-grafana) [inline public/app/plugins/panel/flamegraph/components/FlameGraph/FlameGraph.tsx:97]: Please update todo

File: public/app/features/commandPalette/useMatches.ts (hunk 1 above)
Comment (joshhunt) [reply public/app/features/commandPalette/useMatches.ts:207]: yeah, it probably should have been `haystackItem.length - substringPosition` instead, to prioritise matches at the beginning of the string, but making substringPosition negative works as well.

File: public/app/features/commandPalette/useMatches.ts [hunk 4]
```diff
@@ -201,35 +201,27 @@ function useInternalMatches(filtered: ActionImpl[], search: string): Match[] {
...
+          const score = -substringPosition;
```
Comment (joshhunt) [inline public/app/features/commandPalette/useMatches.ts:207]: i changed this to `substringPosition * -1` so hopefully its easier to catch, and added a comment explaining why we invert it for score

**Review by joshhunt — APPROVED** [review]

**Review by joey-grafana — APPROVED** [review]

**Pull request merged**